src/account/Account.ts

```diff
@@ -206,10 +206,9 @@ export abstract class Account {
    * @category Account (On-Chain Model)
    */
   static fromPrivateKey(args: CreateEd25519AccountFromPrivateKeyArgs): Ed25519Account;
-  static fromPrivateKey(args: CreateEd25519SingleKeyAccountFromPrivateKeyArgs): SingleKeyAccount;
   static fromPrivateKey(args: CreateSingleKeyAccountFromPrivateKeyArgs): SingleKeyAccount;
-  static fromPrivateKey(args: CreateAccountFromPrivateKeyArgs): Account;
-  static fromPrivateKey(args: CreateAccountFromPrivateKeyArgs) {
+  static fromPrivateKey(args: CreateAccountFromPrivateKeyArgs): SingleKeyAccount;
+  static fromPrivateKey(args: CreateAccountFromPrivateKeyArgs): Ed25519Account | SingleKeyAccount {
```

Comment:
I don't think we lose type inference here.  I made this changes to simplify things a bit.  I'll remove from the PR though since it's not core to the PR.

Reply:
Oh I know why.  Because I wanted to change SingleKeyAccount.privateKey to not be of type PrivateKey and be Ed25519PrivateKey | Secp256k1PrivateKey instead.  These are just the downstream changes.